File: producer_pact/src/test/java/com/example/BeerRestBase.java
package com.example;

import java.util.Random;

import io.restassured.config.EncoderConfig;
import io.restassured.module.mockmvc.RestAssuredMockMvc;
import io.restassured.module.mockmvc.config.RestAssuredMockMvcConfig;
import org.junit.jupiter.api.BeforeEach;

public abstract class BeerRestBase {
	//remove::start[]
	ProducerController producerController = new ProducerController(oldEnough());
	StatsController statsController = new StatsController(statsService());

	@BeforeEach
	public void setup() {
		// https://github.com/spring-cloud/spring-cloud-contract/issues/1428
		EncoderConfig encoderConfig = new EncoderConfig().appendDefaultContentCharsetToContentTypeIfUndefined(false);
		RestAssuredMockMvc.config = new RestAssuredMockMvcConfig().encoderConfig(encoderConfig);
		RestAssuredMockMvc.standaloneSetup(this.producerController, this.statsController);
	}

	private PersonCheckingService oldEnough() {
		return argument -> argument.age >= 20;
	}

	private StatsService statsService() {
		return name -> new Random().nextInt();
	}
	//remove::end[]
}

File: producer_kotlin/src/test/kotlin/com/example/BeerRestBase.kt
package com.example

import io.restassured.config.EncoderConfig
import io.restassured.module.mockmvc.RestAssuredMockMvc
import io.restassured.module.mockmvc.config.RestAssuredMockMvcConfig
import org.junit.jupiter.api.BeforeEach
import java.util.*

abstract class BeerRestBase {
    internal var producerController = ProducerController(oldEnough())
    internal var statsController = StatsController(statsService())

    @BeforeEach
    fun setup() {
        val encoderConfig: EncoderConfig = EncoderConfig().appendDefaultContentCharsetToContentTypeIfUndefined(false)
        val restAssuredConf: RestAssuredMockMvcConfig = RestAssuredMockMvcConfig().encoderConfig(encoderConfig)
        RestAssuredMockMvc.config = restAssuredConf
        RestAssuredMockMvc.standaloneSetup(this.producerController, this.statsController)
    }

    private fun oldEnough(): PersonCheckingService {
        return object : PersonCheckingService {
            override fun shouldGetBeer(personToCheck: PersonToCheck): Boolean? {
                return personToCheck.age >= 20
            }
        }
    }

    private fun statsService(): StatsService {
        return object : StatsService {
            override fun findBottlesByName(name: String): Int {
                return Random().nextInt()
            }
        }
    }
}
